src/Scaffolding/Scaffolders/DataObjectScaffolder.php

```
@@ -665,10 +665,13 @@ protected function createFields(Manager $manager)
         }
 
         $resolver = function ($obj, $args, $context, $info) {
-            /**
-             * @var DataObject $obj
-             */
-            $field = StaticSchema::inst()->accessField($obj, $info->fieldName);
+            /* @var DataObject $obj */
+            try {
+                $field = StaticSchema::inst()->accessField($obj, $info->fieldName);
+            } catch (InvalidArgumentException $e) {
```

InvalidArgumentException isn't part of the FieldAccessorInterface API, it's specific to CaseInsensitiveFieldAccessor

Seems like you could be using a NaiveFieldAccessor here as well?

You should be able to use FieldAccessorInterface::getObjectFieldName instead i.e.

```
$accessor = StaticSchema::inst()->getFieldAccessor();
if (!$accessor || !$accessor->getObjectFieldName($obj, $info->fieldName)) {
  return null;
}
$field = StaticSchema::inst()->accessField($obj, $info->fieldName);
```

---

Yup, I think I pointed this out in an edit:

> The chosen option isn't great, but it doesn't break any APIs, should qualify for patch release as it fixes a regression, and it's extremely unlikely that anyone would be using a custom FieldAccessorInterface abstraction.

There aren't any great solutions, here. We could do `accessField($obj, $info->fieldName, [CaseInsensitiveFieldAccessor::GRACEFUL])`, but again, that's implementation-specific code.

Ideally all those `CaseInsensitiveFieldAccessor` options are moved to the interface, and that way there would be a reasonable expectation that its implementors know what to do with those options.

---

What if -- we copy all the options constants in `CaseInsensitiveFieldAccessor` to the interface, and use those instead. Keep the old ones around or backward compat, but just mark that they're no longer used in a comment.

Then, `accessField` with a suppress errors option becomes plausible in the abstraction?

---

> What if -- we copy all the options constants in CaseInsensitiveFieldAccessor to the interface

Then it becomes a minor release

What's wrong with the code I provided above?  Is essentially just `if (obj.hasField($field)) { value = obj.getField($field);}`

---

The issue I see is that `if (!$accessor || !$accessor->getObjectFieldName($obj, $info->fieldName)) {`

Will throw.

---

Oh, I see! I didn't realise that was part of the interface. YES.

Great solution. Thank you.